app/controllers/doctor_patients_controller.rb

```diff
@@ -0,0 +1,8 @@
+class DoctorPatientsController < ApplicationController
+  def destroy
+    doctor_patient = DoctorPatient.find(params[:id])
```

Comment:
The `params[:id]` is not the ID of a `DoctorPatient` object, it is the ID of a `Doctor` object, that you passed in with this code: `<%= button_to "Delete", doctor_patient_path(@doctor, patient), method: :delete %>`. You are deleting the wrong object.